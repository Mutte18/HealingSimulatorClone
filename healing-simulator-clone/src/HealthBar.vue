<template>
  <div class="outerHealthBar" :style="getIsAliveBackgroundColorStatus">
    <div class="innerHealthBar" :style="getHealthPointStyle"></div>
    <div class="healthBarText"> {{displayHealthStatus}} </div>
  </div>
</template>

<script>
    export default {
      props: {
        healthPoints: {
          type: Number,
          required: true
        },
        maxHealth: {
          type: Number,
          required: true
        },
        isAlive: {
          type: Boolean,
          required: true
        }
      },
      data() {
        return {

        }
      },
      methods: {

      },
      computed: {
        getHealthPointStyle() {
          let backgroundColor = '';
          if (this.healthPoints > 75) {
            backgroundColor = 'green';
          } else if (this.healthPoints > 50) {
            backgroundColor = 'lightgreen';
          } else if (this.healthPoints > 25) {
            backgroundColor = 'yellow';
          } else if (this.healthPoints > 0) {
            backgroundColor = 'red';
          }
          return {
            backgroundColor: backgroundColor,
            width: this.healthPoints + '%'
          }
        },
        getIsAliveBackgroundColorStatus(){
          if(this.isAlive){
            return {
              backgroundColor: 'lightgray'
            }
          }
          else {
            return {
              backgroundColor: 'dimgray'
            }
          }
        },
        displayHealthStatus() {
          if(this.isAlive){
            return `${this.healthPoints} /  ${this.maxHealth}`
          }
          else {
            return 'DEAD'
          }
        }
      },
    }
</script>

<style scoped>
  .outerHealthBar{
    width: 100%;
    height: 50px;
    margin: auto 0;
  }

  .healthBarText{
    position: relative;
    text-align: center;
    bottom: 35px;
  }

  .innerHealthBar{
    width: inherit;
    position: relative;
    height: 50px;
    transition: width 500ms;
  }
</style>
