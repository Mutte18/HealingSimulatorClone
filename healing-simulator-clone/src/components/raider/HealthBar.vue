<template>
  <div class="outerHealthBar" :style="getIsAliveBackgroundColorStatus">
    <div class="innerHealthBar" :style="getClassHealthbars"></div>
    <div class="healthBarText">
      <p v-if="isAlive">
        HP: {{  this.healthPoints }} / {{ this.maxHealth }} <br>
        <span class="classification">{{ this.classification }} </span>
      </p>
      <p v-else> DEAD </p>
  </div>
  </div>
</template>

<script>
import { classifications } from '../../raiderClassifications';

export default {
  props: {
    healthPoints: {
      type: Number,
      required: true,
    },
    maxHealth: {
      type: Number,
      required: true,
    },
    isAlive: {
      type: Boolean,
      required: true,
    },
    classification: {
      type: String,
      required: true,
    },
  },
  data() {
    return {

    };
  },
  methods: {
    getHealthPercentage() {
      return (this.healthPoints / this.maxHealth) * 100;
    },
  },
  computed: {
    /*getHealthPointStyle() {
      let backgroundColor = '';
      if (this.getHealthPercentage() > 75) {
        backgroundColor = 'green';
      } else if (this.getHealthPercentage() > 50) {
        backgroundColor = 'lightgreen';
      } else if (this.getHealthPercentage() > 25) {
        backgroundColor = 'yellow';
      } else if (this.getHealthPercentage() > 0) {
        backgroundColor = 'red';
      }
      return {
        backgroundColor,
        width: `${this.getHealthPercentage()}%`,
      };
    },*/
    getClassHealthbars() {
      let backgroundColor = '';
      switch (this.classification) {
        case classifications.TANK:
          backgroundColor = 'saddlebrown';
          break;
        case classifications.HEALER:
          backgroundColor = 'darkgoldenrod';
          break;
        case classifications.DPS:
          backgroundColor = 'green';
          break;
        case classifications.YOU:
          backgroundColor = 'teal';
          break;
      }
      return {
        backgroundColor,
        width: `${this.getHealthPercentage()}%`,
      };
    },
    getIsAliveBackgroundColorStatus() {
      if (this.isAlive) {
        return {
          backgroundColor: 'lightgray',
        };
      }

      return {
        backgroundColor: 'dimgray',
      };
    },
  },
};
</script>

<style scoped>
  .outerHealthBar{
    width: 100%;
    height: 50px;
    display: flex;
    justify-content: center;
    align-items: center;
    position: relative;
  }

  .healthBarText {
    z-index: 0;
  }

  .classification {
    display: flex;
    justify-content: center;
  }

  .innerHealthBar{
    position: absolute;
    top:0;
    left:0;
    height: 100%;
    transition: width 200ms;
    background-color: gold;
  }
</style>
