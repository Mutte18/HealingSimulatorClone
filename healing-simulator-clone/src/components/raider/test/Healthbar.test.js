import { shallowMount } from "@vue/test-utils";
import HealthBar from '../HealthBar.vue';

let wrapper;

/*beforeEach(() => {
  wrapper = shallowMount(HealthBar, {
    propsData: {
      healthPoints: 100,
      maxHealth: 100,
      isAlive: true,
      classification: 'TANK'
    }
  });
});*/
function setUpMount(propsData){
  return shallowMount(HealthBar, {
    propsData: {
      ...propsData
    }
  });
}


afterEach(() => {
  wrapper.destroy();
});

test('renders correctly when alive', () => {
  wrapper = setUpMount({
    healthPoints: 100,
    maxHealth: 100,
    isAlive: true,
    classification: 'TANK'
  });
  expect(wrapper).toMatchSnapshot();
});

test('renders correctly when dead', () => {
  wrapper = setUpMount({
    healthPoints: 100,
    maxHealth: 100,
    isAlive: false,
    classification: 'TANK'
  });
  expect(wrapper).toMatchSnapshot();
});

