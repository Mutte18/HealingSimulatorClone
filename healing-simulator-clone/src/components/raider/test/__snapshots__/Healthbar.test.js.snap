// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when alive 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;

exports[`renders correctly when dead 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
